python-sdk/src/astro/databricks/api_utils.py

```diff
@@ -18,6 +21,47 @@
 log = logging.getLogger(__name__)
 
 
+def delete_secret_scope(scope_name: str, api_client: ApiClient) -> None:
+    """
+    Delete the scope we created to prevent littering the databricks secret store with one-time scopes.
+    :param scope_name: name of scope to delete
+    :param api_client: populated databricks client
+    """
+    secrets = SecretApi(api_client)
+    try:
+        secrets.delete_scope(scope_name)
+    except HTTPError as h:
+        # We don't care if the resource doesn't exist since we're trying to delete it
+        if not h.response.json()["error_code"] == "RESOURCE_DOES_NOT_EXIST":
+            raise h
+
+
+def create_secrets(scope_name: str, filesystem_secrets: Dict[str, str], api_client: ApiClient) -> None:
+    """
+    Uploads secrets to a scope
+    Before we can transfer data from external file sources (s3, GCS, etc.) we first need to upload the relevant
+    secrets to databricks, so we can use them in the autoloader config. This allows us to perform ad-hoc queries
+    that are not dependent on existing settings.
+    :param scope_name: the name of the secret scope. placing all secrets in a single scope makes them easy to
+        delete later
+    :param filesystem_secrets: a dictionary of k,v secrets where the key is the secret name and the value is
+        the secret value
+    :param api_client: The databricks API client that has all necessary credentials
+    """
+    secrets = SecretApi(api_client)
+    secrets.create_scope(
+        scope=scope_name,
+        initial_manage_principal=None,
+        backend_azure_keyvault=None,
+        scope_backend_type=None,
+    )
+
+    for k, v in filesystem_secrets.items():
+        # Add "astro_sdk_" to the front, so we know which secrets are hadoop settings
+        secret_key = "astro_sdk_" + k
+        secrets.put_secret(scope=scope_name, key=secret_key, string_value=v, bytes_value=None)
```

Comment:
Please add unit tests too so that we don't have to fully rely on integration tests

Reply:
Oh weird I thought I had made one. Yes totally.

Reply:
Done